pkg/provider/aws/resources/eks_test.go

```diff
@@ -36,9 +44,9 @@ func Test_CreateEksCluster(t *testing.T) {
 					{Source: "aws:eks_fargate_profile:test-app-test-cluster", Destination: "aws:eks_cluster:test-app-test-cluster"},
 					{Source: "aws:eks_fargate_profile:test-app-test-cluster", Destination: "aws:iam_role:test-app-test-cluster-FargateExecutionRole"},
 					{Source: "aws:eks_fargate_profile:test-app-test-cluster", Destination: subnet.Id()},
-					{Source: "aws:eks_node_group:test-app-test-cluster", Destination: "aws:eks_cluster:test-app-test-cluster"},
-					{Source: "aws:eks_node_group:test-app-test-cluster", Destination: "aws:iam_role:test-app-test-cluster-NodeGroupRole"},
-					{Source: "aws:eks_node_group:test-app-test-cluster", Destination: subnet.Id()},
+					{Source: "aws:eks_node_group:private_t3.medium", Destination: "aws:eks_cluster:test-app-test-cluster"},
+					{Source: "aws:eks_node_group:private_t3.medium", Destination: "aws:iam_role:test-app-test-cluster.private_t3.medium"},
+					{Source: "aws:eks_node_group:private_t3.medium", Destination: subnet.Id()},
```

Comment:
https://github.com/klothoplatform/klotho/blob/ifc2-foundations/pkg/sanitization/aws/eks.go#L24-L37

/shrug

Reply:
right im saying we need to update this. We never had node group sanitization in deploy lib, but looking at our ts code we do remove .s https://github.com/klothoplatform/klotho/blob/main/pkg/infra/pulumi_aws/iac/eks.ts#L539. So we need to update our sanitization here since before we only had a static name we never ran into it